src/autogluon/cloud/job/ray_job.py

```diff
@@ -27,20 +61,79 @@ def info(self) -> Dict:
         dict
             A dictionary containing the general information about the job.
         """
-        raise NotImplementedError
+        assert self.job_name is not None, "No job detected. Please submit a job first"
+        info = dict(name=self.job_name, status=self.get_job_status(), artifact_path=self.get_output_path())
+        return info
 
-    def run(self, **kwargs):
-        """Execute the job"""
-        raise NotImplementedError
+    def run(
+        self,
+        entry_point: str,
+        runtime_env: Dict[str, Any] = None,
+        job_name: Optional[str] = None,
+        wait: bool = True,
+        ray_submit_job_args: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        """
+        Execute the job
+
+        Parameters
+        ----------
+        entry_point: str
+            The shell command to run for this job.
+        runtime_env: Dict[str, Any]. Default None
+            The runtime environment to install and run this job in.
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/running-applications/job-submission/sdk.html
+        job_name: Optional[str]. Default None
+            Name of the job being submitted. If not specified, will create one with pattern `ag-ray-{timestamp}`
+        wait: bool. Default True
+            Whether to wait
+        ray_submit_job_args : Optional[Dict[str, Any]]. Default None
+            Additional args to be passed to ray JobSubmissionClient.submit_job call.
+            To learn more,
+                https://docs.ray.io/en/latest/cluster/running-applications/job-submission/doc/ray.job_submission.JobSubmissionClient.submit_job.html#ray.job_submission.JobSubmissionClient.submit_job
+        """
+        if job_name is None:
+            job_name = f"ag-ray-{get_utc_timestamp_now()}"
+        if ray_submit_job_args is None:
+            ray_submit_job_args = {}
+        self.client.submit_job(
+            entrypoint=entry_point, runtime_env=runtime_env, submission_id=job_name, **ray_submit_job_args
+        )
+        logger.log(20, f"Submitted job {job_name} to the cluster")
+        self._job_name = job_name
+        if wait:
+            logger.warning("We don't support log streaming currently. Logs will be avaialbe when the job is finished")
+            self._wait_until_status(
```

Comment:
Not sure if there is an timeout mechanism on ray job but would be good to add a timeout instead of waiting forever